packages/broker/src/plugins/storage/StorageConfig.ts

```diff
@@ -97,7 +97,10 @@ export class StorageConfig {
 
     async refresh(): Promise<void> {
         const res = await fetch(`${this.apiUrl}/storageNodes/${this.nodeId}/streams`)
-        const json = await res.json()
+        if (!res.ok) {
+            throw new Error(`Refresh failed: ${res.status} ${await res.text()}`)
+        }
+        const json = (await res.json()) || []
```

Comment:
Is this `[]` fallback needed? In which situation it would be applied in practice?

If such error happens, would it be better we e.g. throw an `Error` and therefore ignore the refresh(). Now we remove all storage assignments.

Reply:
Good point.